spec.v1.yml

```diff
@@ -0,0 +1,364 @@
+          description: name of collection
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: bookNumber
+          description: number of the book
+          required: true
+          schema:
+            type: string
+        - $ref: "#/components/parameters/limit"
+        - $ref: "#/components/parameters/page"
+  "/collections/{collectionName}/books/{bookNumber}/hadiths":
+    get:
+      summary: Get the list of hadith of a book for a collection
+      description: ""
+      responses:
+        "200":
+          description: Paginated list of hadiths of a book of a collection
+          content:
+            application/json:
+              schema:
+                type: object
+                allOf:
+                  - properties:
+                      data:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/Hadith"
+                  - $ref: "#/components/schemas/PaginatedResponse"
+      parameters:
+        - in: path
+          name: collectionName
+          description: name of collection
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: bookNumber
+          description: number of the book
+          required: true
+          schema:
+            type: string
+        - $ref: "#/components/parameters/limit"
+        - $ref: "#/components/parameters/page"
+  "/collections/{collectionName}/hadiths/{hadithNumber}":
+    get:
+      summary: Get a hadith of a collection
+      description: ""
+      responses:
+        "200":
+          description: Hadith of a book of a collection
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Hadith"
+      parameters:
+        - in: path
+          name: collectionName
+          description: name of collection
+          required: true
+          schema:
+            type: string
+        - in: path
+          name: hadithNumber
+          description: number of the hadith
+          required: true
+          schema:
+            type: string
+  /hadiths/random:
+    get:
+      summary: Get a randomly selected hadith
+      description: ""
+      responses:
+        "200":
+          description: A randomly selected hadith
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Hadith"
+servers:
+  - url: https://api.sunnah.com/v1/
+components:
+  parameters:
+    limit:
+      in: query
+      name: limit
+      description: Maximum number of items
+      required: false
+      schema:
+        type: integer
+        maximum: 100
+        default: 50
+    page:
+      in: query
+      name: page
+      description: Offset for pagination
+      required: false
+      schema:
+        type: integer
+        default: 1
+  securitySchemes:
+    APIKey:
+      name: X-API-Key
```

Comment:
I think the current implementation looks for the `x-aws-secret` value in the header.

Reply:
No, that's a secret between API Gateway and our server. `X-API-Key` is what API consumers include in their request.